code/script.js

```diff
@@ -1,51 +1,452 @@
+  setTimeout(proposeActivity, 1000);
+}
+
+
+//Iteration 3
+//part 1
+//bot replies back using message proposing activities
+const proposeActivity = () => {
+  showMessage(`Welcome to our hotel Le Banian, ${userName}!`, "bot")
+  showMessage(`This booking service is offered to you directly from your room, at your leisure.`, "bot")
+  setTimeout(() => { showMessage(`Which activity would you like to book today?`, "bot"); }, 2000)
+  setTimeout(hotelActivity, 3000);
+}
+
+//part 2
+//buttons showing for choosing activity
+const hotelActivity = () => {
+  // clear input form
+  nameForm.removeChild(nameInput)
+  nameForm.removeChild(sendButton)
+  // buttons appear for activity choice
+  activityBtn.innerHTML += `
+  <button id="spa" value="spa">Spa</button>
+  <button id="golf" value="golf">Golf</button>
+  <button id="boathouse" value="boathouse">Boathouse</button>
+  <button id="diner" value="diner">Diner</button>
+  `
+  //Using addEventListener method on click event to call next function. "CURRYING" in programming lang.
+  document.getElementById('spa').addEventListener('click', activityConfirmation);
+  document.getElementById('golf').addEventListener('click', activityConfirmation);
+  document.getElementById('boathouse').addEventListener('click', activityConfirmation);
+  document.getElementById('diner').addEventListener('click', activityConfirmation);
+
+  // old code saved- to understand the refactoring/currying
+  // const spaBtn = document.getElementById('spa')
+  // const golfBtn = document.getElementById('golf')
+  // const boathouseBtn = document.getElementById('boathouse')
+  // const dinerBtn = document.getElementById('diner')
+
+  // spaBtn.addEventListener('click', spaActivity);
+  // golfBtn.addEventListener('click', golfActivity);
+  // boathouseBtn.addEventListener('click', boathouseActivity);
+  // dinerBtn.addEventListener('click', dinerReservation);
+}
+
+//Iteration 4
+// Function to confirm chosen activity
+const activityConfirmation = (event) => {
+  event.preventDefault();
+  //targeting the value of the clicked button (event)
+  activitySelection = event.target.value;
+  showMessage(`Option chosen : ${activitySelection}`, "user");
+  setTimeout(() => { showMessage(`Do you confirm your ${activitySelection} choice, ${userName}?`, "bot"); }, 1500)
+  //remove irrelevant buttons
+  activityBtn.innerHTML = "";
+  activityBtn.innerHTML += `
+  <button id= "yes" value= "yes">Yes</button>
+  <button id = "no" value = "no" >No</button > `
+  document.getElementById('yes').addEventListener('click', confirmationAnswer);
+  document.getElementById('no').addEventListener('click', confirmationAnswer);
+}
+
+//Iteration 5
+//part 1
+//If user confirms (answer is yes)
+const confirmationAnswer = (event) => {
+  event.preventDefault();
+  const answer = event.target.value;
+
+  //Ternary not functioning when answer is no, calling activitySelection oavsett -using if else instead, old code:
```

Comment:
Nice that you included this even though it wasn't working as expected. I'm not sure either why it's not working either when looking through it.